python/ray/experimental/channel.py

```diff
@@ -159,3 +160,32 @@ def end_read(self):
         self._worker.core_worker.experimental_mutable_object_read_release(
             [self._base_ref]
         )
+
+    def set_error(self, e: Exception) -> None:
+        """
+        Shutdown the channel with the specified error object. New readers will see
+        the error raised when they try to read from the channel.
+
+        Does not block.
+
+        Args:
+            e: The exception object to write to the channel.
+        """
+        logger.debug(f"Writing error to channel: {self._base_ref}: {e}")
+        serialized_exc = self._worker.get_serialization_context().serialize(e)
+        try:
+            self._worker.core_worker.experimental_mutable_object_put_serialized(
+                serialized_exc,
+                self._base_ref,
+                num_readers=1,
+                try_wait=True,
```

Comment:
is this true here because if write lock is already held, that means the value is already available, so we don't need to write an error to it? (can you comment from code?)